system/Model.php

```diff
@@ -348,6 +348,42 @@ public function find($id = null)
 		return $row['data'];
 	}
 
+	/**
+	 * Fetches the column of database from $this->table with a primary key
+	 * matching $id.
+	 *
+	 * @param mixed|array|null $id One primary key or an array of primary keys
+	 * @param mixed|array|null $column_name Column name or null for primaryKey
+	 *
+	 * @return array|null array of table's column values
+	 */
+	public function findColumn($id = null, string $column_name = null) : array
+	{
+
+		if(is_null($column_name))
+			$column_name = $this->primaryKey;
+
+		if(!is_string($column_name))
```

Comment:
This might be an incorrect check. The type hint enforces it to be either a string or null. So checking if it's a string probably isn't right. Checking if it's null might be better. However, if you don't want to allow a column name parameter at all, then remove the possibility to make it null.

Reply:
I extend my CI4 to support arrays for primaryKey that's why I am checking it here for a string. 

However if null is passed as the 2nd arg and primaryKey is not set or is set to null then ... it is worth to throw exception.

Reply:
Agreed - but my point was that you should probably check if it is null, not if it's a string, since PHP enforces that it will be  string if it's present.